spec/lib/chess/core/board_spec.rb

```diff
@@ -20,56 +22,52 @@
       expect(check).to be(true)
     end
 
-    it 'creates a hash at @pieces' do
-      expect(board.pieces).to be_a(Hash)
-    end
-
     it 'creates a king for white team' do
-      expect(board.pieces[Chess::WHITE_TEAM].find_pieces_of(:K).size).to be(1)
+      expect(board.pieces[Chess::WHITE_TEAM].king.instance_of?(Chess::Core::Pieces::King)).to be(true)
```

Comment:
There is a better matcher for these scenarios than `be`
```suggestion
      expect(board.pieces[Chess::WHITE_TEAM].king).to be_a(Chess::Core::Pieces::King)
```

The same thing applies for all the changes below that expect something to be a certain type of object
https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/type-matchers

Reply:
yeah, done!